app/validation/validator.py

```diff
@@ -47,13 +50,17 @@ def validate(self, user_data):
             logger.debug('Item {} ({}) valid: {}'.format(item_id, type(item), result.is_valid))
             self._validation_store.store_result(item_id, result)
 
-            self._validate_container(item.container)
+            if result.is_valid:
+                self._validate_container(item.container, user_data)
 
         # Return true/False for the set of values
         for item_id in user_data.keys():
             result = self._validation_store.get_result(item_id)
+            item = self._schema.get_item_by_id(item_id)
+            container_result = self._validation_store.get_result(item.container.id)
+
             if result:
-                if not result.is_valid:
+                if not result.is_valid or not container_result.is_valid:
```

Comment:
At this point it's possible for container_result to be None and then the page explodes like this:


![screen shot 2016-05-13 at 12 14 18 3](https://cloud.githubusercontent.com/assets/14093825/15246408/84e4939e-1904-11e6-9011-aea4282d1091.png)